exp/controllers/azureasomanagedcontrolplane_controller_test.go

```diff
@@ -201,6 +218,15 @@ func TestAzureASOManagedControlPlaneReconcile(t *testing.T) {
 				},
 			},
 		}
+		managedCluster := &asocontainerservicev1.ManagedCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "mc",
+				Namespace: cluster.Namespace,
+			},
+			Status: asocontainerservicev1.ManagedCluster_STATUS{
+				Fqdn: ptr.To("endpoint"),
+			},
+		}
```

Comment:
Can we add a test case here for when `managedCluster.Status.PrivateFQDN != nil`?

Reply:
Instead of scaffolding another one of these big test cases for that, would it be good enough to test the `getControlPlaneEndpoint()` function directly?

Reply:
Looks good!